src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -0,0 +1,77 @@
+package it.pagopa.interop.catalogprocess.common.readmodel
+
+import it.pagopa.interop.catalogmanagement.model.CatalogItem
+import it.pagopa.interop.catalogmanagement.model.persistence.JsonFormats._
+import it.pagopa.interop.catalogprocess.api.impl.Converter.convertFromApiDescriptorState
+import it.pagopa.interop.catalogprocess.model.EServiceDescriptorState
+import it.pagopa.interop.commons.cqrs.service.ReadModelService
+import org.mongodb.scala.Document
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates.{`match`, count, project, sort}
+import org.mongodb.scala.model.Filters
+import org.mongodb.scala.model.Projections.{computed, fields, include}
+import org.mongodb.scala.model.Sorts.ascending
+import cats.implicits._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReadModelQueries {
+
+  def listEServices(
+    name: Option[String],
+    producersId: List[String],
+    states: List[EServiceDescriptorState],
+    offset: Int,
+    limit: Int
+  )(readModel: ReadModelService)(implicit ec: ExecutionContext): Future[PaginatedResult[CatalogItem]] = {
+    val query = listEServicesFilters(name, producersId, states)
+
+    for {
+      // Using aggregate to perform case insensitive sorting
+      //   N.B.: Required because DocumentDB does not support collation
+      eServices <- readModel.aggregate[CatalogItem](
+        "eservices",
+        Seq(
+          `match`(query),
+          project(fields(include("data"), computed("lowerName", Document("""{ "$toLower" : "$data.name" }""")))),
+          sort(ascending("lowerName"))
+        ),
+        offset = offset,
+        limit = limit
+      )
+      // Note: This could be obtained using $facet function (avoiding to execute the query twice),
+      //   but it is not supported by DocumentDB
+      count     <- readModel.aggregate[TotalCountResult](
```

Comment:
I was thinking that this could probably go in the `aggregate` function in common, and "tomorrow" be implemented with the `facet` function

Reply:
nvm, this would be feasible with `facet` but it's too tricky in this configuration

Reply:
👍